src/context/ServiceInfoContext.tsx

```diff
@@ -59,6 +61,7 @@ interface JmSessionData {
   offer_list: Offer[] | null
   nickname: string | null
   rescanning: boolean
+  rescan_progress?: number
```

Comment:
Last things before we can merge.. `/session` response does not contain a `rescan_progress` property. Please remove it.

(This causes the scanning text to flicker as the `/session` call updates it to `undefined` and the `getrescaninfo` update to the actual progress number)

Other than that, I think it is ready to be merged.